cmd/rm.go

```diff
@@ -57,20 +58,33 @@ func prune(id string) error {
 		return err
 	}
 
+	// take an exclusive lock on the cache directory
+	cacheDir, err := cache.NewCache(id, user)
+	if err != nil {
+		return err
+	}
+	err = cacheDir.Create()
+	if err != nil {
+		return err
+	}
+	err = cacheDir.Lock(cache.Exclusive)
+	if err != nil {
+		return err
+	}
```

Comment:
`NewCache()` calls `FullImageIdFromPartial()`that can potentially read the cache directory without a lock, also `cacheDir.Create()` creates a directory in the cache without a lock (the lock is requested later `cacheDir.Lock()`)

Reply:
so this is a bit of a chicken or the egg scenario. The source of truth for the fullImageId is currently the list of directories in the cache dir. So, the top level cache directory needs to be read before creating the lock. This is the same behavior as main. There are some options to avoid this but I think they are out of the scope of this PR, e.g. we could refactor the directories to be the shortImageId.

https://github.com/containers/podman-bootc/blob/main/pkg/vm/vm_linux.go#L42-L50

https://github.com/containers/podman-bootc/blob/47a710d620dae3b5b998837ae60504a40f0c5c74/pkg/vm/vm.go#L29

Reply:
Yes, for reading we have a race, reading the dir and acquiring the lock, we can barely work around it, by checking the dir again after the lock. But, the real fix will be to use sqlite to store the list of VMs instead of reading the cache dir